app/services/fastly_config/update.rb

```diff
@@ -0,0 +1,58 @@
+module FastlyConfig
```

Comment:
All of this logic was built on the thinking that the most common case is that we do _not_ update Fastly. Updates are few and far between. That's why I added logic to determine if an update is even required - to save us from several unnecessary API calls during deployment.

There are 1 or 2 spots that could be further optimized to re-use data from the initial check when we go to update. But I'm holding off on that until I see how the other configs shakeout.